test/cri-containerd/container_test.go

```diff
@@ -594,3 +594,185 @@ func Test_RunContainer_NonDefault_User(t *testing.T) {
+			Metadata: &runtime.ContainerMetadata{
+				Name: t.Name() + "-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+			Annotations: map[string]string{
+				"containerd.io/snapshot/io.microsoft.container.storage.reuse-scratch": "true",
+			},
+		},
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	containerID := createContainer(t, client, ctx, request)
+	defer removeContainer(t, client, ctx, containerID)
+	startContainer(t, client, ctx, containerID)
+	defer stopContainer(t, client, ctx, containerID)
+
+	// Verify that we only have one disk mounted. Generally we'd have sda + sdb for two containers
+	// (Pause + 1 workload container).
+	cmd := []string{"ls", "/dev/sdb"}
+	containerExecReq := &runtime.ExecSyncRequest{
+		ContainerId: containerID,
+		Cmd:         cmd,
+		Timeout:     20,
+	}
+	r := execSync(t, client, ctx, containerExecReq)
+	if r.ExitCode == 0 {
+		t.Fatalf("expected /dev/sdb to not be present %d: %s", r.ExitCode, string(r.Stdout))
+	}
+}
+
+func Test_RunContainer_ShareScratch_CheckSize_LCOW(t *testing.T) {
+	requireFeatures(t, featureLCOW)
+
+	pullRequiredLcowImages(t, []string{imageLcowK8sPause, imageLcowAlpine})
+
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sandboxRequest := getRunPodSandboxRequest(t, lcowRuntimeHandler)
+
+	podID := runPodSandbox(t, client, ctx, sandboxRequest)
+	defer removePodSandbox(t, client, ctx, podID)
+	defer stopPodSandbox(t, client, ctx, podID)
+
+	request := &runtime.CreateContainerRequest{
+		PodSandboxId: podID,
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: "",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+			Annotations: map[string]string{
+				"containerd.io/snapshot/io.microsoft.container.storage.reuse-scratch": "true",
+			},
+		},
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	request.Config.Metadata.Name = "Container-1"
+	containerIDOne := createContainer(t, client, ctx, request)
+	defer removeContainer(t, client, ctx, containerIDOne)
+	startContainer(t, client, ctx, containerIDOne)
+	defer stopContainer(t, client, ctx, containerIDOne)
+
+	request.Config.Metadata.Name = "Container-2"
+	containerIDTwo := createContainer(t, client, ctx, request)
+	defer removeContainer(t, client, ctx, containerIDTwo)
+	startContainer(t, client, ctx, containerIDTwo)
+	defer stopContainer(t, client, ctx, containerIDTwo)
+
+	// Write a 10MB file and then check size available on the root fs in both containers.
+	// It should be the same amount as it's being backed by the same disk.
+	cmd := []string{"dd", "if=/dev/urandom", "of=testfile", "bs=10MB", "count=1"}
+	containerExecReq := &runtime.ExecSyncRequest{
+		ContainerId: containerIDOne,
+		Cmd:         cmd,
+		Timeout:     20,
+	}
+	r := execSync(t, client, ctx, containerExecReq)
```

Comment:
you think we want to cache the initial state as well? and in the end in addition to validating that the free space is the same both containers, we also check that it was different before. just to make sure that the write didn't happen somewhere else.

Reply:
Sure that makes sense